### PR TITLE
build(tests): type check Buffer.test.ts

### PR DESCRIPTION
**Pull request by gjf7**
ref #352

Our linter and TypeScript config disagreed on test files. Linter included them, config didn't.

https://github.com/neovim/node-client/blob/65ddb5e7c8fa6c1bd4a6cd6564ad0cdf83b7e92c/packages/neovim/tsconfig.json#L7

https://github.com/neovim/node-client/blob/65ddb5e7c8fa6c1bd4a6cd6564ad0cdf83b7e92c/package.json#L37

This PR should fix this as well.

### CODE REVIEW

**Comment by justinmk** [comment]
Thanks for helping with this! Probably want to do a couple files at a time, not one big PR.

File: packages/neovim/src/api/Buffer.test.ts [hunk 1]
```diff
@@ -216,7 +220,7 @@ describe('Buffer API', () => {
         expect(await buffer.getOption('copyindent')).toBe(false);
 
         // Restore option
-        buffer.setOption('copyindent', initial);
+        buffer.setOption('copyindent', initial!);
```
Comment (justinmk) [inline packages/neovim/src/api/Buffer.test.ts:223]: nit: could at assert(initial) on line 217, that might be a better pattern in general.

File: packages/neovim/tsconfig.json [hunk 2]
```diff
@@ -4,5 +4,5 @@
     "outDir": "./lib"
   },
   "include": ["src"],
-  "exclude": ["node_modules", "__tests__", "**/*.test.ts", "examples", "lib"]
+  "exclude": ["node_modules", "examples", "lib"]
```
Comment (justinmk) [inline packages/neovim/tsconfig.json:7]: to unblock this PR so we can work incrementally, maybe leave the old exclude and add specific `include` for the fixed files, e.g.:

    "include": ["src", "**/api/Buffer.test.ts"],

**Review by justinmk — APPROVED** [review]

File: packages/neovim/src/api/Buffer.test.ts (hunk 1 above)
Comment (gjf7) [reply packages/neovim/src/api/Buffer.test.ts]: Use assert for type narrowing? I didn't know that. Thanks for teaching me something new!

File: packages/neovim/tsconfig.json (hunk 2 above)
Comment (gjf7) [reply packages/neovim/tsconfig.json]: Fixed

**Pull request merged**